README.md

```diff
@@ -14,9 +18,103 @@ Communicate information about background jobs.
 
 1. Run `bundle install` from your shell.
 
+1. Generate migrations so `FrenchToast` can render information when javascript isn't available.
+
+```
+rails generate french_toast:install
+rake db:migrate
+```
+
 ## Usage
 
-TODO
+### General French Toast
+
+Simplest thing you can do is use the default french toast helper in a view
+
+```erb
+# application.html.erb
+<%= french_toast %>
+```
+
+Create a partial in `views/french_toast`
+
+```erb
+# _foo.html.erb
+<div>You'll see me eventually! <%= your_name %></div>
+```
+
+Finally, in a background job call the `notify` method with the template you'd like to use, with an optional payload
+
+```ruby
+#... something_job.rb
+def perform
+  FrenchToast.notify("foo", payload: { your_name: "Jim-bob" })
+end
+```
+
+```javascript
+document.addEventListener('french_toast:notify',
+  function(json) {
+    alert("Your foo is done, " + json["your_name"]);
+  });
+```
+
+### Specifying specific sections
+
+Sometimes you need to be able to specify where content should go.
+
+For this we can namespace the `french_toast` partial
+
+```erb
+# posts/index.html.erb
+<%= french_toast :post %>
+```
+
+You would create the partial the same way, but when calling notify you need to specify the location
+
+```ruby
+#... something_job.rb
+def perform
+  FrenchToast.notify("foo", name: :post, payload: {})
+end
+```
+
+```javascript
+document.addEventListener('french_toast:notify:post',
+  function(json) {
+    alert("Your foo is done, " + json["your_name"]);
+  });
+```
+
+### Don't store notification for synchronous rendering
+
+Say you want to use `FrenchToast` to update a list of posts with the most recent post when it arrives, but you already have server side rendering for the posts, so you only need to render things asynchronously.
+
+That's easy!
```

Comment:
Please avoid words like “easy”, which are subjective and can make people feel bad if they find this hard.